### ISSUE TITLE
Unable to select existing cloud credentials creating an EKS cluster

### ISSUE THREAD

**Issue by ghost** [rancher/rancher#33306]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
Create cloud credentials for AWS in Rancher.
Start to create an EKS cluster.
Cloud credentials can not be selected, Add New fields are displayed.

**Result:**
Create new credentials within the form.
Continue.
Next time the cloud credentials can be selected, but it is not listed in the Cloud Credentials menu utility (only the one created through that is listed).

**Expected Result:**
It should be able to select from the list of existing cloud credentials.
Add new should not be the default action.
Cloud credentials entered (if available later) should be saved or having an option to save to the existing list of credentials.

**Other details that may be helpful:**
EKS cluster failed to create. :) Will play with it and open an other ticket later.

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): master-head 4d18400
- Installation option (single install/HA): Single node k3s

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): Hosted Amazon EKS
- Machine type (cloud/VM/metal) and specifications (CPU/memory): cloud, t3.medium
- Kubernetes version (use `kubectl version`):

```
Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.2", GitCommit:"092fbfbf53427de67cac1e9fa54aaa09a28371d7", GitTreeState:"clean", BuildDate:"2021-06-16T12:59:11Z", GoVersion:"go1.16.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.7+rke2r2", GitCommit:"132a687512d7fb058d0f5890f07d4121b3f0a2e2", GitTreeState:"clean", BuildDate:"2021-05-26T04:47:27Z", GoVersion:"go1.15.8b5", Compiler:"gc", Platform:"linux/amd64"}
```

- Docker version (use `docker version`):

```
n/a
```

**Comment by thedadams** [comment]
Hi @phakkarainen. Thanks for submitting this issue. Question: does refreshing the page cause the new cloud credential to become available?

**Comment by phakkarainen** [comment]
Thanks for the reply.

I would assume refreshing would not have helped, since once not able to see a list of creds in the form, I went back to Cloud Credentials menu item to make sure it is there. It was, went back to create the cluster, and the issue was the same, nothing to choose from. 

I can sure try this next week once back in business to be sure.

Another thing though is: I created new creds inside the Create an EKS Cluster form, and then I had two separate creds; one available in the Create Cluster dropdown and the other one in Cloud Credentials. 

Will report back next Tuesday.
